deeppavlov/models/kbqa/tree_to_sparql.py

```diff
@@ -148,14 +287,17 @@ def find_clause_node(self, root: Node, unknown_branch: Node) -> Tuple[Node]:
     def find_named_entity(self, node: Node, conj_list: List[Node], desc_list: List[Tuple[str, int]],
                                 positions: List[int], cut_clause: bool) -> List[Tuple[str, int]]:
         if node.children:
-            if self.find_nmod_appos:
-                used_desc = [node for node in node.children if node.deprel == "appos"]
+            if self.find_nmod_appos(node, positions):
+                used_desc = [elem for elem in node.children if elem.deprel == "appos"]
             else:
                 used_desc = node.children
             for elem in used_desc:
                 if (not cut_clause or (cut_clause and elem.deprel != "acl")) and elem not in conj_list \
                     and (elem.deprel != "appos" or (elem.deprel == "appos" \
-                    and (not elem.children or (len(elem.children) == 1 and elem.children[0].deprel == "flat:name")))):
+                    and (not elem.children or (len(elem.children) == 1 and elem.children[0].deprel == "flat:name")
+                    or (len(elem.children) > 1 and ("«" in [nd.form for nd in elem.children]
+                    or '"' in [nd.form for nd in elem.children]
+                    or '``' in [nd.form for nd in elem.children]))))):
```

Comment:
This is quite frightening. Could you decompose this condition into multiple `if`s or simplify it in other way? For example, `("«" in [nd.form for nd in elem.children] or '"' in [nd.form for nd in elem.children] or '``' in [nd.form for nd in elem.children])` could replaced with `{"«", '"', '``'} & {nd.form for nd in elem.children}`